app/src/main/java/mozilla/lockbox/store/AutoLockStore.kt

```diff
@@ -49,14 +48,18 @@ class AutoLockStore(
             .map { lockCurrentlyRequired() }
             .subscribe(lockRequired as Subject)
 
-        Observables.combineLatest(lifecycleStore.lifecycleEvents, dataStore.state)
-            .filter { it.first == LifecycleAction.Background && it.second != DataStore.State.Locked }
+        lifecycleStore.lifecycleEvents
+            .filter { it == LifecycleAction.Background }
+            .switchMap { dataStore.state.take(1) }
```

Comment:
I am concerned that this store is becoming too complicated for the thing it's doing. 

Conceptually, I feel it should be emitting events when the auto-lock timer is triggering, and the route presenter should be mapping them into lock actions.

By shoving all this logic here, we're spreading the auto-locking/locking/routing into lots of places which will make things harder to debug in the long run.